sdk/tests/tutorials/valuation/test_identifier_mismatch.py

```diff
@@ -0,0 +1,279 @@
+            )
+            for price in prices
+        ]
+
+        return self.quotes_api.upsert_quotes(
+            scope=self.market_data_scope,
+            request_body={
+                "quote" + str(request_number): requests[request_number]
+                for request_number in range(len(requests))
+            },
+        )
+
+    def create_configuration_recipe(
+        self, recipe_scope, recipe_code, default_id
+    ) -> lusid.models.ConfigurationRecipe:
+        """
+        Creates a configuration recipe that can be used inline or upserted,
+        and sets the default lookup identifier for quotes
+        :param str recipe_scope: The scope for the configuration recipe
+        :param str recipe_code: The code of the the configuration recipe
+        :param str default_id: The default identifier used for looking up quotes
+        :return: ConfigurationRecipe
+        """
+
+        return models.ConfigurationRecipe(
+            scope=recipe_scope,
+            code=recipe_code,
+            market=models.MarketContext(
+                market_rules=[],
+                suppliers=models.MarketContextSuppliers(
+                    equity=self.market_data_provider
+                ),
+                options=models.MarketOptions(
+                    default_supplier=self.market_data_provider,
+                    default_instrument_code_type=default_id,
+                    default_scope=self.market_data_scope,
+                ),
+            ),
+        )
+
+    def test_raises_exception(self) -> None:
+        """
+        Tests that aggregation/valuation raises an 'MarketResolverFailure' exception
+        when quotes and transactions are loaded using different identifiers. This only
+        occurs when the identifiers (in this case 'Isin') have not been added to the
+        instruments used in valuation
+        """
+        # Upsert quotes specifying the quote identifier
+        quotes_response = self.upsert_quotes(instrument_id="Isin")
+        assert len(quotes_response.failed) == 0
+        # Set default look-up id for quotes
+        in_line_recipe = self.create_configuration_recipe(
+            self.recipe_scope, self.recipe_code, default_id="Isin"
+        )
+
+        # Call aggregation with recipe identifiers
+        aggregation_request = self.create_aggregation_request(in_line_recipe)
+
+        with self.assertRaises(Exception) as context:
+            # Complete aggregation
+            self.aggregation_api.get_aggregation(
+                scope=TestDataUtilities.tutorials_scope,
+                code=self.portfolio_code,
+                aggregation_request=aggregation_request,
+            )
+            # Check that 'MarketResolverFailure' exception raised
+            self.assertEqual(
+                json.loads(context.exception.body)["name"], "MarketResolverFailure"
+            )
+
+    def test_valuation_with_id_mismatch(self) -> None:
```

Comment:
maybe rename `id_mismatch` to something `different_quote_and_transaction_instrument_ids`, i think "mismatch" makes it sound like a problem